src/main/java/com/example/sixthmafiabot/validators/UserValidator.java

```diff
@@ -16,71 +17,73 @@
 import java.util.concurrent.CompletableFuture;
 
 @Component
+@Slf4j
 public class UserValidator extends BaseValidator {
 
     @Autowired
     UserRepository userRepository;
 
     @Async("asyncExecutor")
-    public void validateUser(User user) throws ValidationException{
-        Set<ConstraintViolation<User>> violations =
+    public void validateUser(UserDTO user) throws ValidationException{
+        Set<ConstraintViolation<UserDTO>> violations =
                 validator.validate(user);
 
         if (!violations.isEmpty()) {
             StringBuilder sb = new StringBuilder();
-            for (ConstraintViolation<User> constraintViolation : violations) {
+            for (ConstraintViolation<UserDTO> constraintViolation : violations) {
                 sb.append(constraintViolation.getMessage());
             }
             throw new ValidationException("Error occurred: " + sb.toString());
         }
     }
 
     @Async("asyncExecutor")
-    public void validateIfAlreadyExists(User user) throws AlreadyExistsExcepeion {
+    public void validateIfAlreadyExists(UserDTO user) throws AlreadyExistsExcepeion {
 
         boolean alreadyExists =  userRepository
                 .getUserByTelegramId(user.getTelegramId())
                 .join() != null;
 
         if(alreadyExists){
+
             throw new AlreadyExistsExcepeion("User with identifier = %s already exists"
                     .formatted(user.getTelegramId())
             );
         }
-
     }
 
     @Async("asyncExecutor")
-    public CompletableFuture<User> validateCreate(User user) throws ServiceValidationError {
+    public CompletableFuture<UserDTO> validateCreate(UserDTO userDTO) throws ServiceValidationError {
 
         try{
-            validateIfAlreadyExists(user);
+            validateIfAlreadyExists(userDTO);
         }
         catch (AlreadyExistsExcepeion ex){
             throw new ServiceValidationError(ex.getMessage());
+
         }
 
         try {
-            validateUser(user);
+            validateUser(userDTO);
         }
         catch (ValidationException ex){
             throw new ServiceValidationError(ex.getMessage());
         }
 
-        return CompletableFuture.completedFuture(user);
+        return CompletableFuture.completedFuture(userDTO);
     }
 
     @Async("asyncExecutor")
-    public CompletableFuture<User> validateUpdate(User user) throws ServiceValidationError {
+    public CompletableFuture<UserDTO> validateUpdate(UserDTO userDTO) throws ServiceValidationError {
 
```

Comment:
add validateAlreadyExists, because you can update user's telegramId to used one, or do not allow to update telegramId
also, use different DTO for create and update, because not all fields are updatable

Reply:
+